consensus/polybft/signer/public.go

```diff
@@ -49,49 +25,55 @@ func (p *PublicKey) MarshalJSON() ([]byte, error) {
 }
 
 // UnmarshalJSON implements the json.Marshaler interface.
-func (p *PublicKey) UnmarshalJSON(b []byte) error {
-	var d []byte
+func (p *PublicKey) UnmarshalJSON(jsonBytes []byte) error {
```

Comment:
Why do we need the `UnmarshalJSON` here? I think it should suffice to use `UnmarshalText` here, the same goes for `MarshalJSON` and `MarshalText`.

Reply:
It is implemented because of validators snapshots (because we are persisting snapshots as JSON representation in bolt database): https://github.com/0xPolygon/polygon-edge/pull/938#issuecomment-1323486276.

On the other hand, it is worth trying to switch to `(Un)MarshalText` as I haven't tried it.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1161/commits/fa7eba62eebaae0527dd0cb54840e4d4dc53134e